sphinx/texinputs/sphinx.sty

```diff
@@ -682,13 +710,28 @@
 % for captions of literal blocks
 % also define `\theH...` macros for hyperref
 \newcounter{literalblock}
+\ifspx@opt@nonumfigreset
+  \def\theliteralblock {\arabic{literalblock}}
+  \def\theHliteralblock{\arabic{literalblock}}
+\else
+\ifx\spx@opt@numfigreset\@empty
```

Comment:
in fact this branch will be never taken because the writers/latex.py will create sphinx.sty package option nonumfigreset or numfigreset=section, chapter,... I will revise there, I think what I did initially was in case numfig config setting is not set, then no package option is passed to sphinx so this branch is then taken. But I made a last minute change in latex.py so currently this is not the case.

Reply:
I have modified numfig handling logic in latex.py in second commit, so now indeed this legacy branch will be taken if numfig is False, thus same LaTeX macros as formerly will get executed in that case.

In passing I noticed that with ``'howto'`` the numbering of code-blocks is not coherent with the one of figures and tables, the latter are numbered by default continously, but code-blocks are reset at sections.